### ISSUE TITLE
unescaped unaligned output

### ISSUE THREAD

**Issue by daurnimator** [leegao/see.lua#4]
```
> require"see"(_G)
.rawget(?)          .error(?)           .dofile(?)          .utf8[6]
.setmetatable(?)    ._VERSION = "Lua 5.3"                   .select(?)
.loadstring(?)      .pcall(?)           .math[35]           .collectgarbage(?)
._PROMPT2 = ">> "        .require(?)         .getmetatable(?)
.print(?)           .arg[1]             ._G[42]             .unpack(?)
.loadfile(?)        .rawlen(?)          ._PROMPT = "> "
.tostring(?)        .bit32[12]          .p(o, prefix)       .debug[16]
.module(?)          .pairs(?)           .package[10]        .rawset(?)
.coroutine[7]       .string[17]         .os[11]             .type(?)
.xpcall(?)          .load(?)            .table[8]           .tonumber(?)
.rawequal(?)        .io[14]             .assert(?)          .next(?)
.ipairs(?)
```
![screenshot_2017-01-05_08-53-45](https://cloud.githubusercontent.com/assets/127971/21660675/7babd9ec-d324-11e6-911c-d8c7f183f59e.png)

where: `_PROMPT="\1\27[32;1m\2>\1\27[0m\2 "` and `_PROMPT2="\1\27[36;1m\2>>\1\27[0m\2 "`
and the escape sequences *are* output directly.

You should probably escape the output (in 5.3 you can use `string.format("%q")`) so that you don't accidently print terminal escape sequences (or other control codes).

**Comment by leegao** [comment]
👍 thanks for the report, I'll look into it

**Comment by leegao** [comment]
I just pushed a change to master that addresses this:

![Imgur](http://i.imgur.com/Ok8bK3U.png)

Thanks for the quality bug reports :)

**Comment by daurnimator** [comment]
Unfortunately `"%q"` isn't good enough in lua 5.1.

Looking at your commit:

  - you don't need `tostring` if it's already a string
  - In 5.3 `%q` happily takes a number, or `nil` as well as a string

**Comment by leegao** [comment]
Thanks for the code-review, I have a further question, would

```lua
local function escape(str)
    -- replace non-ascii characters in str with their \xxx control code
    local output = ''
    for i = 1, #str do
        local byte = str:byte(i)
        if byte < 32 or byte > 126 then
            output = output .. ('\\%d'):format(byte)
        else
            output = output .. string.char(byte)
        end
    end
    return output
end
```
be an idiomatic/efficient way to replace non-ascii characters with their control sequence equivalent?

**Comment by daurnimator** [comment]
Not really.
One thing to remember in lua is to aviod concatenating in a loop (build up a table and use `table.concat` instead).

However for you purpose here you can use `string.gsub`.
`output = str:gsub("%c", function(c) return string.format("\\%03d", string.byte(c)) end)`

**Comment by leegao** [comment]
Thanks for all the help :) I just pushed up a change that wraps all '%q' formatting calls and escapes the corresponding output. This feels really hacky/clowny and it doesn't really feel like the best solution, but I'm not sure what would be better.

**Comment by daurnimator** [comment]
> This feels really hacky/clowny and it doesn't really feel like the best solution, but I'm not sure what would be better.

Hence why it was fixed in Lua 5.3.... but see.lua supports 5.1, so you have to do the silly-ness :P

This can be made easier with [compat-5.3](https://github.com/keplerproject/lua-compat-5.3)

**Comment by leegao** [comment]
By the way, I really appreciate all of the free help :) If you ever need anything, don't hesitate to ask
